<launch>
<!-- Simple launch file to run offline optic flow on recorderd usb cam images-->
  <arg name="oflow_plot" default="false" doc="plot tangential optic flow (not filtered) vs gamma along time"/>


  <node pkg="nodelet" type="nodelet" name="standalone_nodelet" args="manager" output="screen" />

  <node name="rgb_optic_flow_node" pkg="optic_flow_node" type="optic_flow_node" output="screen" >
      <remap from="/usb_cam/image_raw" to="/videofile/image_raw"/>
      <param name="image_center_x" value="320"/>
      <param name="image_center_y" value="240"/>
      <param name="inner_ring_radius" value="220"/>
      <param name="ring_dr" value="10"/>
      <param name="num_ring_points" value="160"/>
      <param name="num_rings" value="5"/>
      <param name="pixel_scale" value="215"/>
      <param name="alpha" value="0.7"/>

      <param name="num_ring_points" value="100"/>
      <param name="num_rings" value="100"/>

      <param name="num_of_rows" value="23"/>
      <param name="num_of_cols" value="23"/>

      <param name="pyr_window_size" value="12"/>
      <param name="num_of_pyramids" value="3"/>
      <!-- <param name="pyr_window_size" value="40"/>
      <param name="num_of_pyramids" value="3"/> -->

      <param name="if_blur" value="false"/>
      <!-- Blur size must be an int odd number -->
      <param name="blur_size" value="21"/>

      <param name="resize_of_cols" value="0.0625"/>
      <param name="resize_of_rows" value="0.0625"/>
      <param name="resize_vz_cols" value="0.5000"/>
      <param name="resize_vz_rows" value="0.5000"/>

  </node>

  <node name="thermal_optic_flow_node" pkg="optic_flow_node" type="optic_flow_node" output="screen" >
      <remap from="/usb_cam/image_raw" to="/lepton_viz"/>
      <param name="image_center_x" value="320"/>
      <param name="image_center_y" value="240"/>
      <param name="inner_ring_radius" value="220"/>
      <param name="ring_dr" value="10"/>
      <param name="num_ring_points" value="160"/>
      <param name="num_rings" value="5"/>
      <param name="pyr_window_size" value="41"/>
      <param name="pixel_scale" value="215"/>
      <param name="alpha" value="0.7"/>

      <param name="num_ring_points" value="100"/>
      <param name="num_rings" value="100"/>

      <param name="num_of_rows" value="11"/>
      <param name="num_of_cols" value="11"/>

      <param name="pyr_window_size" value="8"/>
      <param name="num_of_pyramids" value="3"/>

      <param name="if_blur" value="false"/>
      <!-- Blur size must be an int odd number -->
      <param name="blur_size" value="41"/>

      <param name="resize_of_cols" value="0.1250"/>
      <param name="resize_of_rows" value="0.1250"/>
      <param name="resize_vz_cols" value="1.0000"/>
      <param name="resize_vz_rows" value="1.0000"/>


  </node>

</launch>
